### ISSUE TITLE
Validation DataLoader returns zero batches during Inference

### ISSUE THREAD

**Issue by liee2333** [QuantumLab-ZY/HamGNN#50]
Dear developer of HamGNN
            When using HamGNN for training, the data loader returned an empty batch during the validation set check phase. The full error is as follows:
```
File "/share/home/.../pytorch_lightning/trainer/data_loading.py", line 508, in _reset_eval_dataloader
  if has_len_all_ranks(dataloader, self.training_type_plugin, module)
pytorch_lightning.utilities.exceptions.MisconfigurationException: 
Total length of `Dataloader` across ranks is zero. Please make sure that it returns at least 1 batch.
```
config.yaml context
```
dataset_params:
  batch_size: 2
  split_file: null
  test_ratio: 0.1
  train_ratio: 0.8
  val_ratio: 0.1
  graph_data_path: /share/home/liyuting/TEST/DeePH/work_dir/inference/1_O4Si2/OUT.silica  # Directory where graph_data.npz is located

losses_metrics:
  losses:
  - loss_weight: 27.211
    metric: mae
    prediction: hamiltonian
    target: hamiltonian
  - loss_weight: 0.01  # 添加带能损失，权重为 0.01
    metric: mae
    prediction: band_energy
    target: band_energy
  metrics:
  - metric: mae
    prediction: hamiltonian
    target: hamiltonian
  - metric: mae
    prediction: band_energy
    target: band_energy

optim_params:
  lr: 0.0001  # 调整学习率为 0.0001
  lr_decay: 0.5
  lr_patience: 5
  gradient_clip_val: 1.0
  max_epochs: 3000
  min_epochs: 100
  stop_patience: 30

output_nets:
  output_module: HamGNN_out
  HamGNN_out:
    ham_only: true
    ham_type: abacus
    nao_max: 13
    add_H0: True
    symmetrize: True
    calculate_band_energy: True  # 启用带能计算
    num_k: 5  # 设置随机生成的 K 点数量
    band_num_control: 8  # 设置考虑的带数
    k_path: null  # 允许随机采样 K 点
    soc_switch: False
    nonlinearity_type: gate
    spin_constrained: False
    collinear_spin: False
    minMagneticMoment: 0.5

profiler_params:
  progress_bar_refresh_rat: 1
  train_dir: /share/home/liyuting/TEST/DeePH/work_dir/inference/1_O4Si2/OUT.silica/pre

representation_nets:
  HamGNN_pre:
    cutoff: 26.0
    cutoff_func: cos
    edge_sh_normalization: component
    edge_sh_normalize: true
    irreps_edge_sh: 0e + 1o + 2e + 3o + 4e + 5o
    irreps_node_features: 64x0e+64x0o+32x1o+16x1e+12x2o+25x2e+18x3o+9x3e+4x4o+9x4e+4x5o+4x5e+2x6e
    num_layers: 3
    num_radial: 64
    num_types: 100
    rbf_func: bessel
    set_features: true
    num_heads: 4
    radial_MLP: [64, 64]
    correlation: 2
    num_hidden_features: 16
    use_kan: False
    radius_scale: 1.01
    build_internal_graph: False

setup:
  GNN_Net: HamGNNpre
  accelerator: null
  ignore_warnings: true
  checkpoint_path: /share/home/liyuting/TEST/DeePH/work_dir/save_dir/version_2/band_checkpoints/version_7/checkpoints/epoch=26-val_loss =0.000000.ckpt # 设置预训练模型权重路径
  load_from_checkpoint: False  # 启用加载预训练权重
  resume: False
  num_gpus: null
  precision: 32
  property: hamiltonian
  tage: test  
```
Looking forward to your assistance
Thank you.
Best regards,
liyuting

**Comment by QuantumLab-ZY** [comment]
Hi liyuting,

Please check the directory at `/share/home/liyuting/TEST/DeePH/work_dir/inference/1_O4Si2/OUT.silica` to see if the file `graph_data.npz` is present. Additionally, verify that the file size is as expected to ensure that `graph_data.npz` is not empty.

Best,
Yang
